cg/meta/upload/scoutapi.py

```diff
@@ -1,69 +1,100 @@
+"""File includes api to uploading data into Scout"""
 # -*- coding: utf-8 -*-
 import logging
 from pathlib import Path
 
-from cg.apps import hk, scoutapi, madeline
-from cg.meta.analysis import AnalysisAPI
-from cg.store import models, Store
 from ruamel import yaml
 
+from cg.apps import hk, madeline, scoutapi
+from cg.meta.analysis import AnalysisAPI
+from cg.store import Store, models
+
 LOG = logging.getLogger(__name__)
 
 
-class UploadScoutAPI(object):
+class UploadScoutAPI:
+    """Class that handles everything that has to do with uploading to Scout"""
 
-    def __init__(self, status_api: Store, hk_api: hk.HousekeeperAPI,
-                 scout_api: scoutapi.ScoutAPI,
-                 analysis_api: AnalysisAPI, madeline_exe: str, madeline=madeline,
-                 ):
+    def __init__(
+        self,
+        status_api: Store,
+        hk_api: hk.HousekeeperAPI,
+        scout_api: scoutapi.ScoutAPI,
+        analysis_api: AnalysisAPI,
+        madeline_exe: str,
+        madeline=madeline,
+    ):
         self.status = status_api
         self.housekeeper = hk_api
         self.scout = scout_api
         self.madeline_exe = madeline_exe
         self.madeline = madeline
         self.analysis = analysis_api
 
+    def build_samples(self, analysis_obj: models.Analysis, hk_version_id: int = None):
+        """Loop over the samples in an analysis and build dicts from them"""
+
+        for link_obj in analysis_obj.family.links:
+            sample_id = link_obj.sample.internal_id
+            bam_tags = ["bam", sample_id]
+            bam_file = self.housekeeper.files(
+                version=hk_version_id, tags=bam_tags
+            ).first()
+            bam_path = bam_file.full_path if bam_file else None
+            mt_bam_tags = ["bam-mt", sample_id]
+            mt_bam_file = self.housekeeper.files(
+                version=hk_version_id, tags=mt_bam_tags
+            ).first()
+            mt_bam_path = mt_bam_file.full_path if mt_bam_file else None
+            vcf2cytosure_tags = ["vcf2cytosure", sample_id]
+            vcf2cytosure_file = self.housekeeper.files(
+                version=hk_version_id, tags=vcf2cytosure_tags
+            ).first()
+            vcf2cytosure_path = (
+                vcf2cytosure_file.full_path if vcf2cytosure_file else None
+            )
+            sample = {
+                "analysis_type": link_obj.sample.application_version.application.analysis_type,
+                "sample_id": sample_id,
+                "capture_kit": None,
```

Comment:
Why is capture kit set to None?

Reply:
No idea, I have not touched that code. It was Blacked

Reply:
Aha I moved the code into a new function but did not change anything

Reply:
If you don't need it in Scout, why bother with it at all here? I think we could remove it if there is no need for it in Scout..

Reply:
but I agree to it being done later